src/workspace/internal/workspaceuiarrangmentstream.cpp

```diff
@@ -0,0 +1,105 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "workspaceuiarrangmentstream.h"
+
+#include "global/xmlreader.h"
+#include "global/xmlwriter.h"
+
+#include "workspacetypes.h"
+
+using namespace mu::workspace;
+using namespace mu::framework;
+using namespace mu::system;
+
+static const std::string UI_ARRANGMENT_TAG("UiArrangment");
+static const std::string UI_SETTINGS_TAG("Settings");
+static const std::string UI_SETTINGS_NAME_TAG("name");
+
+AbstractDataPtrList WorkspaceUiArrangmentStream::read(IODevice& sourceDevice) const
```

Comment:
Seems that this class is a full copy of WorkspaceSettingsStream (the only difference is in UI_SETTINGS_TAG). I suggest using WorkspaceSettingsStream to read UI arrangements from a workspace. You can add WorkspaceTag (or some other enum that contains Settings and UiArrangement values)  to WorkspaceSettingsStream constructor and resolve the required XML tag inside that class